Vertexvis/vertex-web-sdk
# [PLAT-3145] Scene-tree loading improvements

## Summary

Makes a few improvements to the loading spinner in the scene-tree element:
- Prevents a case that was causing multiple `GetTree` requests when metadata was used
- Fixes an issue where attempting to filter the tree (without metadata) would result in an infinite spinner
- Fixes a couple cases where the loading spinner could get "stuck" and never resolve when initially loading the tree
- Fixes an issue where connecting/reconnecting the tree too quickly could cause infinite reconnection attempts
- Fixes an issue where the subscription timeout would invalidate pages without a re-fetch, and cause the tree to be stuck in a loading state
- Fixes an issue where letting the scene tree go idle (long enough to hit the timeout at 4 minutes) would then result in ~2s interval reconnects

## Test Plan

- Verify the fixes listed above
- Verify the issue in https://vertexvis.atlassian.net/browse/PLAT-2763 no longer reproduces
  - The fix for this issue was backed out to fix another issue with reconnection, but I'm not seeing different behavior afterward

## Release Notes

N/A

## Possible Regressions

https://vertexvis.atlassian.net/browse/PLAT-2763, scene tree connections

## Dependencies

N/A
